crates/uv-tool/src/lib.rs

```diff
@@ -106,15 +109,15 @@ impl InstalledTools {
     }
 
     /// Lock the tools directory.
-    fn acquire_lock(&self) -> Result<LockedFile, Error> {
+    pub fn acquire_lock(&self) -> Result<LockedFile, Error> {
         Ok(LockedFile::acquire(
             self.root.join(".lock"),
             self.root.user_display(),
         )?)
     }
 
-    /// Lock a tool directory.
-    fn acquire_tool_lock(&self, name: &str) -> Result<LockedFile, Error> {
+    /// Lock a specific tool directory.
+    pub fn acquire_tool_lock(&self, name: &str) -> Result<LockedFile, Error> {
```

Comment:
We don't actually use this yet. In what case does this makes sense?

Reply:
I think this would make sense for `tool run`, `tool install`, `tool uninstall`, right?

Reply:
But then `tool list` could try to inspect a tool mid-installation.

Reply:
Can probably just always-lock then.